### ISSUE TITLE
Terraform crash when S3 bucket website created with an empty string assigned to index_document 

### ISSUE THREAD

**Issue by jwieringa** [hashicorp/terraform#6253]
### Terraform Version

```
$ terraform -v
Terraform v0.6.14
```

### Affected Resource(s)

- aws_s3_bucket

### Terraform Configuration Files

```hcl
provider "aws" {
  region     = "us-east-1"
  access_key = "${var.aws_access_key}"
  secret_key = "${var.aws_secret_key}"
}

resource "aws_s3_bucket" "v1" {
  bucket = "test-tf-website"

  website {
    index_document = ""
    error_document = ""
  }
}
```

### Working Terraform Configuration Files

```hcl
provider "aws" {
  region     = "us-east-1"
  access_key = "${var.aws_access_key}"
  secret_key = "${var.aws_secret_key}"
}

resource "aws_s3_bucket" "v1" {
  bucket = "test-tf-website"

  website {
    # do not pass an empty string to index_document
    index_document = "index.html"
    error_document = ""
  }
}
```

### Panic Output

Terraform `crash.log` https://gist.github.com/jwieringa/f1e8f851274cd56571460aa73ba52ec7

### Expected Behavior

I expected a S3 bucket to be created with a website configuration containing empty values for `index_document` and `error_document`.

```
$ terraform plan
Refreshing Terraform state prior to plan...

+ aws_s3_bucket.v1
    acl:              "" => "private"
    arn:              "" => "<computed>"
    bucket:           "" => "test-tf-website"
    force_destroy:    "" => "0"
    hosted_zone_id:   "" => "<computed>"
    region:           "" => "<computed>"
    website.#:        "" => "1"
    website_domain:   "" => "<computed>"
    website_endpoint: "" => "<computed>"


Plan: 1 to add, 0 to change, 0 to destroy.
```

### Actual Behavior

In the above configurations, terraform crashed and no bucket was created.

I observed a different behavior against my production configurations which I have not been able to reproduce in an example; however, I think it is worth noting because the `terraform.tfstate` file changed caused subsequent `apply`'s to fail.

+ The bucket for the website, `static_assets`, was not created
+ Bucket, `logs`, not related to the config was removed from `terraform.tfstate`
+ Bucket `logs` was not deleted from AWS
+ Bucket `logs` was not listed in the plan
+ Terraform did not crash, instead it emitted two errors

```
2 error(s) occurred:

* aws_s3_bucket.logs: unexpected EOF
* aws_s3_bucket.static_assets: unexpected EOF
```

To resolve, I recovered a previous version of `terraform.tfstate` from backups.

### Steps to Reproduce

Using the above configurations:

1. `terraform apply`

### References

This issue was similar to the error received on my production configurations.

- https://github.com/hashicorp/terraform/issues/4706

**Comment by catsby** [comment]
Thanks @jwieringa – I've patched this in #6269 ... not that the configuration expects either `index_document` or `redirect_all_requests_to` to be valid, and with #6269 you'll now get an error

**Comment by jwieringa** [comment]
Nice, thank you! :+1:
